models/commit_status.go

```diff
@@ -61,6 +59,7 @@ type CommitStatus struct {
 	SHA         string            `xorm:"VARCHAR(64) NOT NULL INDEX UNIQUE(repo_sha_index)"`
 	TargetURL   string            `xorm:"TEXT"`
 	Description string            `xorm:"TEXT"`
+	ContextHash string            `xorm:"varchar(40) index"`
```

Comment:
What do you think about refactoring Context `TEXT` to `VARCHAR` to remove ContextHash?

Reply:
What's the best size of `context` column?

Reply:
`191` I think.

Reply:
191 is too large since there are so many status record on this table. I think smaller size of context will give us a better performance on group. That's why I create a hash column to do that.

Reply:
‘ci/drone-continuous-integration’ is 31 chars so what about make it 50 max?

Reply:
If performance is really going to be a concern, wouldn't hashing + `BINARY(40)` be better? Or maybe even a lookup table with the different possible Contexts (either by org, repo, or simply globally) - I wouldn't expect to see *many* different Contexts, this would let you group simply by the foreign key.

Reply:
Not a big fan of putting restrictions on this. We often append OS, processor architecture and other things to the name which can make it quite long. (Look at github.com/mono/mono for some example names on any PR)

Reply:
@dr-consit my first version of this PR, I have added a new table to store that different context. But I consider it again that it's unnecessary and we just need an extra hash column. Maybe `CHAR(40)` is better that `varchar(40)` and I'm not familiar with `binary(40)` that may not with good readable when operating database directly. 
@filipnavara in github.com/mono/mono, the max length of context is 39. :)
@techknowlogick maybe `char(64)` is better, but I haven't found any clue about the context length limitation on github's status document.

Reply:
If there is no "actual" (per some standard/doc) limit on the Context size, I would argue that using a hash is much preferable to imposing an "arbitrary" limit.
If using sha-1 as the hashing algorithm (which has a digest size of 160 bits) the correct size of a `BINARY` column would be 20 not 40 as I wrote earlier, which is part of the reason why `BINARY` *should* be faster/better for indexing/grouping.

Having considered the extra table unnecessary, because a hash column was considered better/good enough, does not, I believe, work as an argument for discarding the extra table solution if you reconsider the hash column. I mean the fallback should not automatically become storing the value directly in the status table.
If the better choice between storing in lookup and hash was hash, and you discard hash because of limitations/functionality, then I think you should reconsider lookup.

I still believe that if you want to store the actual Context values, then a lookup table is the better solution. Maybe indexed by Org or Repo, if you expect to see a lot of variance in values between orgs/repos.